broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -155,16 +155,19 @@ private void takeTransientSnapshotInternal(
     }
 
     final var snapshotIndexedEntry = optionalIndexed.get();
-    final Optional<TransientSnapshot> transientSnapshot =
+    final var transientSnapshot =
         constructableSnapshotStore.newTransientSnapshot(
             snapshotIndexedEntry.index(),
             snapshotIndexedEntry.term(),
             lowerBoundSnapshotPosition,
             exportedPosition);
 
     // Now takeSnapshot result can be either true, false or error.
-    transientSnapshot.ifPresentOrElse(
-        snapshot -> takeSnapshot(snapshot, future), () -> future.complete(Optional.empty()));
+    if (transientSnapshot.isLeft()) {
```

Comment:
:thought_balloon: Either supports as well  `ifRightOrLeft`

Reply:
I found this more readable. With `ifRightOrLeft` there are too many lambdas.